src/backend/_scss/application.scss

```diff
@@ -1386,36 +1389,48 @@ a {
     }
 }
 
-.search-filter {
-  display: inline;
-  float: right;
+
+.search-filter-schedule {
+  flex-grow: 1;
   padding-top: 12px;
 
   .fossasia-filter {
-    float: left;
-    font-family: sans-serif;
+    height: 70%;
+    width: 97%;
   }
 
+}
+
+.search-filter-room {
+  flex-grow: 2;
+  padding-left: 2px;
+  padding-top: 12px;
+
   input {
-    line-height: 24px;
+    line-height: 30px;
+    width: 100%;
   }
 
   @media only screen and (max-width: 600px) {
-    display: inline;
-    float: none;
-    width: 40%;
-
     .fossasia-filter {
+      border-radius: 5px;
       float: none;
-      margin: 0 auto;
-      width: 100%;
       height: 33px;
-      border-radius: 5px;
+      margin: 0 auto;
     }
   }
 
 }
 
+.filter-margin {
+  margin-right: 10px;
+}
+
+.navbar-flex {
+  display: flex;
```

Comment:
I tried other approaches but flexbox seemed to work the best for all screen sizes!! The properties like flex-grow and flex-wrap are very useful. That's why I went ahead with it. Thanks!!